Fixed Date picker getBestDateTimePattern bug

sjanic opened the pull request:
When inflating a layout with a DatePicker in Robolectric, we get an IllegalArgumentException with a bad character pattern (see issue https://github.com/robolectric/robolectric/issues/2024). 

This pull request fixes the bug based on the solution from jshvarts in the above mention issue, and includes a test that exposes the bug.

jongerrish commented on robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java:
```diff
@@ -15,8 +21,27 @@
-    assertThat(ICU.getBestDateTimePattern("hm", null)).isEqualTo("h:mm a");
-  }
+    @Test
+    public void getBestDateTimePattern_returnsReasonableValue() {
```
Can you make it two indents like the rest of the project then I'll merge.

jongerrish commented on robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java:
```diff
@@ -15,8 +21,27 @@
     Build.VERSION_CODES.LOLLIPOP })
```
I think we should add LOLLIPOP_MR1 + M to the list of tests run.

sjanic commented:
I've fixed the indent, and added LOLLIPOP_MR1 to the sdk. I don't know how to add the M sdk to the config attribute, which is why it's missing.

Please let me know, if you how to add the M sdk or if I have to do anything else. :)

jongerrish commented on robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java:
```diff
@@ -1,22 +1,47 @@
 @Config(sdk = {
-    Build.VERSION_CODES.LOLLIPOP })
+  Build.VERSION_CODES.LOLLIPOP,
+  Build.VERSION_CODES.LOLLIPOP_MR1})
```
Add Build.VERSION_CODES.M here also

sjanic replied:
I'm getting a compile error that M doesn't exist. The M sdk is installed on my computer (API 23), and I have tried to rerun install-dependencies.rb with the M sdk installed.

Do I need to do anything else, in order for Build.VERSION_CODES to generate the M entry in its class?

jongerrish replied:
AH, no my bad... we don't support M yet, I'm just adding it in the branch :-) This is great as is. Thanks.

jongerrish commented:
...and then merge in master,
